### ISSUE TITLE
Can not use more than one .boto config with BOTO_CONFIG env. variable

### ISSUE THREAD

**Issue by katrielt** [GoogleCloudPlatform/gsutil#281]
I am trying to use two service accounts by using BOTO_CONFIG to point to different .boto files.
When using the first services account, a simple "gsutil ls" works. When trying to switch to the second service account, I will always get an error:
**AccessDeniedException: 403 Forbidden**

I found out that when deleting "rm .gsutil/credstore", I could switch to the second account, but then the 1st one would not work.
After a bit of digging, it seems that the problem happens because the same key is used when caching credentials in **gslib/gcs_json_api.py**, lines 198-200:
```
self.credentials.set_store(
    multistore_file.get_credential_storage_custom_string_key(
        GetCredentialStoreFilename(), self.api_version))
```
Using self.api_version means that the same key index is always used, no matter which service account I use.

**Comment by thobrla** [comment]
Thanks for the report.  Will look into getting it fixed for gsutil 4.14.

**Comment by thobrla** [comment]
This should be fixed for 4.14 with https://github.com/GoogleCloudPlatform/gsutil/commit/98caaff874abeb097f222f1fb00e42d5089961bf

Thanks again for bringing this up.
